### ISSUE TITLE
Managing inheritance in revapi?

### ISSUE THREAD

**Issue by surli** [revapi/revapi#62]
I'm trying to setup revapi for a java project containing big hierarchies, then when I add a method to an interface I obtain traces like that:

```
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtAnnotationMethod<T>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtAnnotationType<T extends java.lang.annotation.Annotation>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtAnonymousExecutable: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtClass<T>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtConstructor<T>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtElement: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtEnum<T extends java.lang.Enum<?>>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtEnumValue<T>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtExecutable<R>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtField<T>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtFormalTypeDeclarer: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtInterface<T>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtMethod<T>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtModifiable: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtMultiTypedElement: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtNamedElement: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtPackage: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtParameter<T>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtType<T>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtTypeMember: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtTypeParameter: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtTypedElement<T>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.declaration.CtVariable<T>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.reference.CtArrayTypeReference<T>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.reference.CtCatchVariableReference<T>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.reference.CtExecutableReference<T>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.reference.CtFieldReference<T>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.reference.CtIntersectionTypeReference<T>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.reference.CtLocalVariableReference<T>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.reference.CtPackageReference: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.reference.CtParameterReference<T>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.reference.CtReference: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.reference.CtTypeParameterReference: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.reference.CtTypeReference<T>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.reference.CtUnboundVariableReference<T>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.reference.CtVariableReference<T>: Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>) @ spoon.reflect.reference.CtWildcardReference: Method was added to an interface. (breaks semantic versioning)
```

Cause around 40 interfaces extends another one in which I added a specific method. Is it possible to make revapi more precise and to target only the interface which changed?

**Comment by metlos** [comment]
With the fix in, this is the complete set of differences found in `spoon-core`:
```
[ERROR] Failed to execute goal org.revapi:revapi-maven-plugin:0.8.1-SNAPSHOT:check (revapi-check) on project spoon-core: The following API problems caused the build to fail:
[ERROR] java.class.added: interface spoon.reflect.visitor.chain.CtConsumableFunction<T extends java.lang.Object>: Class was added. (breaks semantic versioning)
[ERROR] java.class.added: interface spoon.reflect.visitor.chain.CtConsumer<T extends java.lang.Object>: Class was added. (breaks semantic versioning)
[ERROR] java.method.returnTypeTypeParametersChanged: method spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQuery::failurePolicy(spoon.reflect.visitor.chain.QueryFailurePolicy): The return type changed from 'spoon.reflect.visitor.chain.CtQuery<O extends java.lang.Object>' to 'spoon.reflect.visitor.chain.CtQuery'. (breaks semantic versioning)
[ERROR] java.method.returnTypeTypeParametersChanged: method <R extends spoon.reflect.declaration.CtElement> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQuery::filterChildren(spoon.reflect.visitor.Filter<R>): The return type changed from 'spoon.reflect.visitor.chain.CtQuery<T extends spoon.reflect.declaration.CtElement>' to 'spoon.reflect.visitor.chain.CtQuery'. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <R> void spoon.reflect.visitor.chain.CtQuery::forEach(spoon.reflect.visitor.chain.CtConsumer<R>): Method was added to an interface. (breaks semantic versioning)
[ERROR] java.generics.elementNowParameterized: method <R> java.util.List<R> spoon.reflect.visitor.chain.CtQuery::list(): Element now defines formal type parameters. (breaks semantic versioning)
[ERROR] java.generics.formalTypeParameterAdded: method <R> java.util.List<R> spoon.reflect.visitor.chain.CtQuery::list(): A new formal type parameter added to element: 'R'. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <R> java.util.List<R> spoon.reflect.visitor.chain.CtQuery::list(java.lang.Class<R>): Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.returnTypeTypeParametersChanged: method <I, R> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQuery::map(spoon.reflect.visitor.chain.CtFunction<I, R>): The return type changed from 'spoon.reflect.visitor.chain.CtQuery<R extends java.lang.Object>' to 'spoon.reflect.visitor.chain.CtQuery'. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQuery::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>): Method was added to an interface. (breaks semantic versioning)
[ERROR] java.method.returnTypeTypeParametersChanged: method spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQuery::name(java.lang.String): The return type changed from 'spoon.reflect.visitor.chain.CtQuery<O extends java.lang.Object>' to 'spoon.reflect.visitor.chain.CtQuery'. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <T extends spoon.reflect.visitor.chain.CtQuery> T spoon.reflect.visitor.chain.CtQuery::setInput(java.lang.Object[]): Method was added to an interface. (breaks semantic versioning)
[ERROR] java.generics.formalTypeParameterRemoved: interface spoon.reflect.visitor.chain.CtQuery: The formal type parameter 'O' was removed from the element. (breaks semantic versioning)
[ERROR] java.generics.formalTypeParameterRemoved: method void spoon.reflect.visitor.chain.CtQueryImpl::<init>(java.lang.Object[]): The formal type parameter 'T' was removed from the element. (breaks semantic versioning)
[ERROR] java.method.removed: method void spoon.reflect.visitor.chain.CtQueryImpl<O>::<init>(): Method was removed. (breaks semantic versioning)
[ERROR] java.method.returnTypeChangedCovariantly: method spoon.reflect.visitor.chain.CtQueryImpl spoon.reflect.visitor.chain.CtQueryImpl::addInput(java.lang.Object[]): The return type changed covarianly from 'spoon.reflect.visitor.chain.CtQuery<O extends java.lang.Object>' to 'spoon.reflect.visitor.chain.CtQueryImpl'. (breaks semantic versioning)
[ERROR] java.method.added: method <I, R> void spoon.reflect.visitor.chain.CtQueryImpl::evaluate(I, spoon.reflect.visitor.chain.CtConsumer<R>): Method was added. (breaks semantic versioning)
[ERROR] java.method.returnTypeChangedCovariantly: method spoon.reflect.visitor.chain.CtQueryImpl spoon.reflect.visitor.chain.CtQueryImpl::failurePolicy(spoon.reflect.visitor.chain.QueryFailurePolicy): The return type changed covarianly from 'spoon.reflect.visitor.chain.CtQuery<O extends java.lang.Object>' to 'spoon.reflect.visitor.chain.CtQueryImpl'. (breaks semantic versioning)
[ERROR] java.method.returnTypeChangedCovariantly: method <R extends spoon.reflect.declaration.CtElement> spoon.reflect.visitor.chain.CtQueryImpl spoon.reflect.visitor.chain.CtQueryImpl::filterChildren(spoon.reflect.visitor.Filter<R>): The return type changed covarianly from 'spoon.reflect.visitor.chain.CtQuery<T extends spoon.reflect.declaration.CtElement>' to 'spoon.reflect.visitor.chain.CtQueryImpl'. (breaks semantic versioning)
[ERROR] java.method.removed: method int spoon.reflect.visitor.chain.CtQueryImpl<O>::getLength(): Method was removed. (breaks semantic versioning)
[ERROR] java.generics.elementNowParameterized: method <R> java.util.List<R> spoon.reflect.visitor.chain.CtQueryImpl::list(): Element now defines formal type parameters. (breaks semantic versioning)
[ERROR] java.generics.formalTypeParameterAdded: method <R> java.util.List<R> spoon.reflect.visitor.chain.CtQueryImpl::list(): A new formal type parameter added to element: 'R'. (breaks semantic versioning)
[ERROR] java.method.added: method <R> java.util.List<R> spoon.reflect.visitor.chain.CtQueryImpl::list(java.lang.Class<R>): Method was added. (breaks semantic versioning)
[ERROR] java.method.returnTypeChangedCovariantly: method spoon.reflect.visitor.chain.CtQueryImpl spoon.reflect.visitor.chain.CtQueryImpl::logging(boolean): The return type changed covarianly from 'spoon.reflect.visitor.chain.CtQuery<O extends java.lang.Object>' to 'spoon.reflect.visitor.chain.CtQueryImpl'. (breaks semantic versioning)
[ERROR] java.method.returnTypeChangedCovariantly: method <I, R> spoon.reflect.visitor.chain.CtQueryImpl spoon.reflect.visitor.chain.CtQueryImpl::map(spoon.reflect.visitor.chain.CtFunction<I, R>): The return type changed covarianly from 'spoon.reflect.visitor.chain.CtQuery<R extends java.lang.Object>' to 'spoon.reflect.visitor.chain.CtQueryImpl'. (breaks semantic versioning)
[ERROR] java.method.added: method <I> spoon.reflect.visitor.chain.CtQueryImpl spoon.reflect.visitor.chain.CtQueryImpl::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>): Method was added. (breaks semantic versioning)
[ERROR] java.method.returnTypeChangedCovariantly: method spoon.reflect.visitor.chain.CtQueryImpl spoon.reflect.visitor.chain.CtQueryImpl::name(java.lang.String): The return type changed covarianly from 'spoon.reflect.visitor.chain.CtQuery<O extends java.lang.Object>' to 'spoon.reflect.visitor.chain.CtQueryImpl'. (breaks semantic versioning)
[ERROR] java.method.returnTypeChangedCovariantly: method spoon.reflect.visitor.chain.CtQueryImpl spoon.reflect.visitor.chain.CtQueryImpl::setInput(java.lang.Object[]): The return type changed covarianly from 'spoon.reflect.visitor.chain.CtQuery<O extends java.lang.Object>' to 'spoon.reflect.visitor.chain.CtQueryImpl'. (breaks semantic versioning)
[ERROR] java.method.added: method spoon.reflect.visitor.chain.CtQueryImpl spoon.reflect.visitor.chain.CtQueryImpl::stepFailurePolicy(spoon.reflect.visitor.chain.QueryFailurePolicy): Method was added. (breaks semantic versioning)
[ERROR] java.class.nowImplementsInterface: class spoon.reflect.visitor.chain.CtQueryImpl: Class now implements interface 'spoon.reflect.visitor.chain.CtQuery'. (breaks semantic versioning)
[ERROR] java.generics.formalTypeParameterRemoved: class spoon.reflect.visitor.chain.CtQueryImpl: The formal type parameter 'O' was removed from the element. (breaks semantic versioning)
[ERROR] java.class.superTypeTypeParametersChanged: class spoon.reflect.visitor.chain.CtQueryImpl: Super type's type parameters changed from 'spoon.reflect.visitor.chain.CtQuery<O extends java.lang.Object>' to 'spoon.reflect.visitor.chain.CtQuery'. (breaks semantic versioning)
[ERROR] java.method.returnTypeTypeParametersChanged: method <R extends spoon.reflect.declaration.CtElement> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::filterChildren(spoon.reflect.visitor.Filter<R>): The return type changed from 'spoon.reflect.visitor.chain.CtQuery<T extends spoon.reflect.declaration.CtElement>' to 'spoon.reflect.visitor.chain.CtQuery'. (breaks semantic versioning)
[ERROR] java.method.returnTypeTypeParametersChanged: method <I, R> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtFunction<I, R>): The return type changed from 'spoon.reflect.visitor.chain.CtQuery<R extends java.lang.Object>' to 'spoon.reflect.visitor.chain.CtQuery'. (breaks semantic versioning)
[ERROR] java.method.addedToInterface: method <I> spoon.reflect.visitor.chain.CtQuery spoon.reflect.visitor.chain.CtQueryable::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>): Method was added to an interface. (breaks semantic versioning)
[ERROR] java.class.added: class spoon.reflect.visitor.filter.OverriddenMethodQuery: Class was added. (breaks semantic versioning)
[ERROR] java.class.nonPublicPartOfAPI: class spoon.support.compiler.jdt.JDTTreeBuilderHelper: Class 'spoon.support.compiler.jdt.JDTTreeBuilderHelper' is indirectly included in the API (by the means of method return type for example) but the class is not accessible (neither public nor protected).
[ERROR] Use chain of the type in the old API: spoon.support.compiler.jdt.JDTTreeBuilderHelper is returned from method spoon.support.compiler.jdt.JDTTreeBuilderHelper spoon.support.compiler.jdt.JDTTreeBuilder::getHelper() (method spoon.support.compiler.jdt.JDTTreeBuilderHelper spoon.support.compiler.jdt.JDTTreeBuilder::getHelper() is part of the API)
[ERROR] Use chain of the type in the new API: spoon.support.compiler.jdt.JDTTreeBuilderHelper is returned from method spoon.support.compiler.jdt.JDTTreeBuilderHelper spoon.support.compiler.jdt.JDTTreeBuilder::getHelper() (method spoon.support.compiler.jdt.JDTTreeBuilderHelper spoon.support.compiler.jdt.JDTTreeBuilder::getHelper() is part of the API)
[ERROR] java.field.serialVersionUIDUnchanged: field spoon.support.reflect.declaration.CtElementImpl.serialVersionUID: The class changed in an incompatible way with regards to serialization but the serialVersionUID field stayed unchanged. This might be ok and/or desired but is suspicious. (breaks semantic versioning)
[ERROR] java.method.returnTypeTypeParametersChanged: method <P extends spoon.reflect.declaration.CtElement> spoon.reflect.visitor.chain.CtQuery spoon.support.reflect.declaration.CtElementImpl::filterChildren(spoon.reflect.visitor.Filter<P>): The return type changed from 'spoon.reflect.visitor.chain.CtQuery<P extends spoon.reflect.declaration.CtElement>' to 'spoon.reflect.visitor.chain.CtQuery'. (breaks semantic versioning)
[ERROR] java.method.returnTypeTypeParametersChanged: method <I, R> spoon.reflect.visitor.chain.CtQuery spoon.support.reflect.declaration.CtElementImpl::map(spoon.reflect.visitor.chain.CtFunction<I, R>): The return type changed from 'spoon.reflect.visitor.chain.CtQuery<R extends java.lang.Object>' to 'spoon.reflect.visitor.chain.CtQuery'. (breaks semantic versioning)
[ERROR] java.method.added: method <I> spoon.reflect.visitor.chain.CtQuery spoon.support.reflect.declaration.CtElementImpl::map(spoon.reflect.visitor.chain.CtConsumableFunction<I>): Method was added. (breaks semantic versioning)
```

**Comment by surli** [comment]
Seems good :)

**Comment by metlos** [comment]
Fixed in `revapi-java-0.13.1`.

**Comment by surli** [comment]
Thanks!
